File: app/lib/models/goal.dart
class GoalModel {
  final int? id;
  final int userId;
  final String name;
  final double targetAmount;
  final double savedAmount;
  final String? targetDate;
  final String? createdAt;
  final double suggestedMonthlyContribution;

  GoalModel({
    this.id,
    required this.userId,
    required this.name,
    required this.targetAmount,
    required this.savedAmount,
    this.targetDate,
    this.createdAt,
    required this.suggestedMonthlyContribution,
  });

  factory GoalModel.fromJson(Map<String, dynamic> json) {
    return GoalModel(
      id: json['id'] as int?,
      userId: (json['userId'] as num?)?.toInt() ?? 1,
      name: json['name'] as String,
      targetAmount: (json['targetAmount'] as num).toDouble(),
      savedAmount: (json['savedAmount'] as num?)?.toDouble() ?? 0.0,
      targetDate: json['targetDate'] as String?,
      createdAt: json['createdAt'] as String?,
      suggestedMonthlyContribution: (json['suggestedMonthlyContribution'] as num?)?.toDouble() ?? 0.0,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (id != null) 'id': id,
      'userId': userId,
      'name': name,
      'targetAmount': targetAmount,
      'savedAmount': savedAmount,
      if (targetDate != null) 'targetDate': targetDate,
    };
  }
}

File: backend/src/main/java/com/expensetracker/model/Goal.java
package com.expensetracker.model;

import java.math.BigDecimal;
import java.sql.Date;
import java.sql.Timestamp;

public class Goal {
    private Long id;
    private Long userId;
    private String name;
    private BigDecimal targetAmount;
    private BigDecimal savedAmount;
    private Date targetDate;
    private Timestamp createdAt;

    // Calculated helper
    private BigDecimal suggestedMonthlyContribution;

    public Goal() {}

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public Long getUserId() { return userId; }
    public void setUserId(Long userId) { this.userId = userId; }

    public String getName() { return name; }
    public void setName(String name) { this.name = name; }

    public BigDecimal getTargetAmount() { return targetAmount; }
    public void setTargetAmount(BigDecimal targetAmount) { this.targetAmount = targetAmount; }

    public BigDecimal getSavedAmount() { return savedAmount; }
    public void setSavedAmount(BigDecimal savedAmount) { this.savedAmount = savedAmount; }

    public Date getTargetDate() { return targetDate; }
    public void setTargetDate(Date targetDate) { this.targetDate = targetDate; }

    public Timestamp getCreatedAt() { return createdAt; }
    public void setCreatedAt(Timestamp createdAt) { this.createdAt = createdAt; }

    public BigDecimal getSuggestedMonthlyContribution() { return suggestedMonthlyContribution; }
    public void setSuggestedMonthlyContribution(BigDecimal suggestedMonthlyContribution) { this.suggestedMonthlyContribution = suggestedMonthlyContribution; }
}
